debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/IncrementalSnapshotContext.java

```diff
@@ -82,7 +82,7 @@ public IncrementalSnapshotContext(boolean useCatalogBeforeSchema) {
     }
 
     public boolean openWindow(String id) {
-        if (!id.startsWith(currentChunkId)) {
+        if (currentChunkId != null && !id.startsWith(currentChunkId)) {
```

Comment:
I was getting a few unusual `NullPointerException` errors during shutdown of the connector and from what I was able to see this method was being called during shutdown the `currentChunkId` was null; hence why I added the not null check.